### ISSUE TITLE
[Issue]: /Devices Endpoint No Longer Accessible With API Key

### ISSUE THREAD

**Issue by eglia** [jellyfin/jellyfin#11632]
### Please describe your bug

Accessing the "/Decices " API endpoint with an API key fails:
`curl https://jellyfin.mydomain.com/Devices?api_key=myapikey`:
```
HTTP/2 400
Error processing request.
```

The same endpoint works with authorization copied from the web interface:
`curl https://jellyfin.mydomain.com/Devices -H 'Authorization: MediaBrowser Client="Jellyfin Web", Device="Firefox", DeviceId="Foo", Version="10.9.1", Token="Bar"'`
```
HTTP/2 200
{...}
```

The issue did not happen in 10.8.13.

### Reproduction Steps

1. `curl https://jellyfin.mydomain.com/Devices?api_key=myapikey`
2. Get response `Error processing request.`

### Jellyfin Version

10.9.0

### Environment

```markdown
- OS: Debian 12
- Linux Kernel: 6.6.13+bpo-amd64
- Virtualization: Docker
- Clients: n.a.
- Browser: n.a.
- FFmpeg Version: n.a.
- Playback Method: n.a.
- Hardware Acceleration: n.a.
- GPU Model: n.a.
- Plugins:
AudioDB 10.9.1.0
Intro Skipper 0.2.0.3
LDAP-Auth 19.0.0.0
MusicBrainz 10.9.1.0
OMDb 10.9.1.0
Playback Reporting 14.0.0.0
Reports 17.0.0.0
Session Cleaner 3.0.0.0
Studio Images 10.9.1.0
Subtitle Extract 4.0.0.0
TMDb 10.9.1.0
TMDb Box Sets 11.0.0.0
TV Maze 5.0.0.0
TheTVDB 11.0.0.0
Webhook 13.0.0.0
- Reverse Proxy: Caddy
- Base URL: none
- Networking: Macvlan
- Storage: local
```

### Jellyfin logs

```shell
[14:35:57] [ERR] [40] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL GET /Devices.
System.ArgumentException: Guid can't be empty (Parameter 'id')
   at Jellyfin.Server.Implementations.Users.UserManager.GetUserById(Guid id)
   at Jellyfin.Server.Implementations.Devices.DeviceManager.GetDevicesForUser(Nullable`1 userId)
   at Jellyfin.Server.Implementations.Devices.DeviceManager.GetDevicesForUser(Nullable`1 userId)
   at Jellyfin.Api.Controllers.DevicesController.GetDevices(Nullable`1 userId)
   at lambda_method902(Closure, Object)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Jellyfin.Api.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Api.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Api.Middleware.IPBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Api.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Api.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Jellyfin.Api.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
   at Jellyfin.Api.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gnattu** [comment]
You need now to explicitly provide a `userId` in the query string.

**Comment by eglia** [comment]
You are right, with the userID it works as expected. I'm not sure why this is required now, but I guess it does not really matter for me.
But the [API Doc](https://api.jellyfin.org/#tag/Devices) should be updated accordingly.

**Comment by crobibero** [comment]
Linked a PR that will fix this issue

**Comment by crobibero** [comment]
Closing as this will be fixed in 10.9.2.
